app/models/service_template_ansible_playbook.rb

```diff
@@ -151,4 +151,17 @@ def update_catalog_item(options, auth_user = nil)
     end
     super
   end
+
+  def destroy(auth_user = 'system')
+    config_info = options[:config_info]
+    [:provision, :retirement, :reconfigure].each do |action|
+      next unless config_info[action] && config_info[action][:playbook_id]
+      info = config_info[action]
+      playbook = ManageIQ::Providers::EmbeddedAnsible::AutomationManager::Playbook.find(info[:playbook_id])
+      manager_ref = job_template(action).manager_ref
+      ManageIQ::Providers::EmbeddedAnsible::AutomationManager::ConfigurationScript
+        .delete_in_provider_queue(playbook.manager.id, { :manager_ref => manager_ref }, auth_user)
+    end
+    super()
```

Comment:
There is pre-destroy hook. I don't know whether it is the better way to place the logic there. That's the reason I suggest we research the Rails way to clean up resources not through association. 
@Fryguy @kbrock need your inputs.